packages/workspace/src/browser/workspace-trust-service.ts

```diff
@@ -0,0 +1,115 @@
+/********************************************************************************
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { Dialog, StorageService } from '@theia/core/lib/browser';
+import { FrontendApplicationConfigProvider } from '@theia/core/lib/browser/frontend-application-config-provider';
+import { PreferenceService } from '@theia/core/lib/browser/preferences/preference-service';
+import { MessageService } from '@theia/core/lib/common/message-service';
+import { nls } from '@theia/core/lib/common/nls';
+import { Deferred } from '@theia/core/lib/common/promise-util';
+import { inject, injectable, postConstruct } from '@theia/core/shared/inversify';
+import { WorkspaceService } from '.';
+import {
+    WorkspaceTrustPreferences, WORKSPACE_TRUST_EMPTY_WINDOW, WORKSPACE_TRUST_ENABLED, WORKSPACE_TRUST_STARTUP_PROMPT, WORKSPACE_TRUST_STARTUP_PROMPT_NEVER,
+    WORKSPACE_TRUST_STARTUP_PROMPT_ONCE
+} from './workspace-trust-preferences';
+
+const CONFIG_AUTO_TRUST = 'autoTrust';
+const STORAGE_TRUSTED = 'trusted';
+
+@injectable()
+export class WorkspaceTrustService {
+    @inject(WorkspaceService)
+    protected readonly workspaceService: WorkspaceService;
+
+    @inject(PreferenceService)
+    protected readonly preferences: PreferenceService;
+
+    @inject(StorageService)
+    protected readonly storage: StorageService;
+
+    @inject(MessageService)
+    protected readonly messageService: MessageService;
+
+    @inject(WorkspaceTrustPreferences)
+    protected readonly workspaceTrustPref: WorkspaceTrustPreferences;
+
+    protected trusted = new Deferred<boolean>();
+    protected _isTrusted: boolean | undefined;
+
+    @postConstruct()
+    protected async init(): Promise<void> {
+        await this.workspaceService.ready;
+        this.trusted.promise.then(trusted => this._isTrusted = trusted);
+        const implicitTrust = await this.calculateWorkspaceTrust();
+        if (implicitTrust !== undefined) {
+            this.trusted.resolve(implicitTrust);
+        }
+    }
+
+    get isTrusted(): boolean | undefined {
+        return this._isTrusted;
+    }
+
+    get onTrusted(): Promise<boolean> {
+        return this.trusted.promise;
+    }
```

Comment:
The `on__` name convention is usually reserved for events. The way this is used has some similarity to an event (fire once, check often), but I'd prefer either just changing this to `get isTrusted` (since that isn't currently used) or something like
```typescript
getWorkspaceTrust(): Promise<Boolean> {
    return this.trusted.promise;
}
}

Reply:
You are right, I made a mistake there. Fixed it and went with the getWorkspaceTrust approach.